feat: EPI-1131: Add isEdited field to MAR and set default user ID on back-end

huytranbes opened the pull request:
### Changes

_Add a brief description of the changes in this PR to help give the reviewer context._

### Deploys

- [ ] **Deploy to Tamanu Internal** <!-- #deploy -->

### Tests

- [ ] **Run E2E Tests** <!-- #e2e -->

### Remember to...

- ...write or update tests
- ...add UI screenshots and **testing notes** to the Linear issue
- ...add any **manual upgrade steps** to the Linear issue
- ...update the [config reference](https://beyond-essential.slab.com/posts/reference-config-file-0c70ukly), [settings reference](https://beyond-essential.slab.com/posts/reference-settings-0blw1x2q), or any [relevant runbook(s)](https://beyond-essential.slab.com/topics/runbooks-bs04ml6c)
- ...call out additions or changes to **config files** for the deployment team to take note of

<!-- Thank you! -->

huytranbes committed:
refactor(medication): Simplify ChangeStatusModal and Mar components by removing unused props

- Removed `isFuture`, `isPast`, and `selectedDate` props from ChangeStatusModal, MarDetails, and MarStatus components to streamline functionality.
- Updated validation logic in ChangeStatusModal to remove dependency on `isFuture`, enhancing clarity in time slot handling.
- Set `givenTime` to null in ChangeStatusModal for improved data handling during medication administration.

huytranbes committed:
refactor(medication): Update validation for givenTime in ChangeStatusModal and EditAdministrationRecordModal

- Changed the validation type for `givenTime` from date to string and made it nullable in both ChangeStatusModal and EditAdministrationRecordModal to improve data handling during medication administration.
- This adjustment enhances the flexibility of the validation logic, ensuring better compatibility with user input.

huytranbes committed:
feat(medication): Add documentation and model updates for isEdited field in MedicationAdministrationRecord

- Introduced documentation for the new `isEdited` field, indicating whether a medication administration record has been manually edited post-creation.
- Updated the YAML model to include the `isEdited` field with a boolean data type and added a not-null validation test, enhancing tracking of record modifications.

huytranbes committed:
feat(medication): Enhance ChangeStatusModal and EditAdministrationRecordModal with warning modals and improved validation

- Added state management for warning modals in ChangeStatusModal and EditAdministrationRecordModal to handle dose mismatches.
- Refactored initial values setup in ChangeStatusModal for better clarity and maintainability.
- Updated field labels in both modals to utilize TranslatedText for improved localization support.
- Enhanced validation logic to ensure accurate tracking of medication administration records.

huytranbes committed:
feat(medication): Add isEdited field to MedicationAdministrationRecord and update related components

- Introduced `isEdited` boolean field in the MedicationAdministrationRecord model and migration to track modifications.
- Updated API routes to handle optional user IDs for `givenByUserId` and `recordedByUserId`, enhancing flexibility in user tracking.
- Modified MarStatus and related components to incorporate the new `isEdited` field, improving the visibility of record modifications.
- Enhanced validation logic in medication administration workflows to ensure accurate tracking of edits.

huytranbes committed:
Merge branch 'epic-medication-enhancements-and-mar-3' of https://github.com/beyondessential/tamanu into fix/epi-1133-build-change-administration-record-status-workflow-2

khoi-bes commented on packages/facility-server/app/routes/apiv1/medication.js:
```diff
@@ -491,20 +491,26 @@ medication.put(
+      }
     }
 
     record.status = ADMINISTRATION_STATUS.GIVEN;
```
should update record.recordedByUserId in line 517th as well

huytranbes committed:
Merge branch 'epic-medication-enhancements-and-mar-3' of https://github.com/beyondessential/tamanu into fix/epi-1133-build-change-administration-record-status-workflow-2

huytranbes committed:
fix(medication): Update recordedByUserId assignment and track record edits

- Set recordedByUserId to req.user.id if not provided.
- Introduced isEdited flag to indicate if a record has been modified after initial entry.

huytranbes committed:
tweak: not generating MAR when frequency is as directed

huytranbes committed:
misc: uncapitalize 'Edit Administration Record' title
